crypto/ed25519.go

```diff
@@ -74,10 +75,7 @@ func GeneratePrivateKey() (PrivateKey, error) {
 // PublicKey returns a PublicKey associated with the Ed25519 PrivateKey p.
 // The PublicKey is the last 32 bytes of p.
 func (p PrivateKey) PublicKey() PublicKey {
-	rpk := p[32:] // privateKey == private|public
-	var pk PublicKey
-	copy(pk[:], rpk)
-	return pk
+	return PublicKey(p[PublicKeyLen:])
```

Comment:
Not to be too pedantic (this is the last change), I would use `PrivateKeyLen` here because that is what we are actually skipping.

I think I suggested the wrong change so this is on me.